calicoctl/commands/resources.go

```diff
@@ -124,6 +124,16 @@ func getResourceFromArguments(args map[string]interface{}) (unversioned.Resource
 			p.Metadata.CIDR = *cidr
 		}
 		return *p, nil
+	case "bgpPeer":
+		p := api.NewBGPPeer()
+		if name != "" {
+			err := p.Metadata.PeerIP.UnmarshalText([]byte(name))
+			if err != nil {
+				return nil, err
+			}
+		}
+		p.Metadata.Hostname = hostname
```

Comment:
Yeah... the IP version is the bit that messed this up for pools as well.

Fine if this doesn't happen in the alpha timeframe, but I think we want to do this for GA.  It probably means that the way we handle identifying information isn't quite right.